lib/dao.js

```diff
@@ -840,6 +840,11 @@ DataAccessObject.replaceOrCreate = function replaceOrCreate(data, options, cb) {
     return this.create(data, options, cb);
   }
 
+  var forceId = this.settings.forceId;
+  if (forceId && id) {
+    return this.replaceById(data.id, data, options, cb);
```

Comment:
why `data.id`; L838 already set `id` to the value of PK; so the right code would be:
```
return self.replaceById(id, data, options, cb);
```